parser_test.go

```diff
@@ -4156,14 +4156,15 @@ func TestKeywordsNotAllowed(t *testing.T) {
 
 	for keyword := range keywordsNotAllowed {
 		ast, err := Parse(fmt.Sprintf("select %s from t", keyword))
-		require.NoError(t, err)
+		require.Error(t, err)
 		require.Len(t, ast.Errors, 1)
 
 		var e *ErrKeywordIsNotAllowed
 		require.ErrorAs(t, ast.Errors[0], &e)
 		if errors.As(ast.Errors[0], &e) {
 			require.Equal(t, keyword, e.Keyword)
 		}
+		require.ErrorAs(t, err, &e)
```

Comment:
The other new style could be checking the error with `err` in the case, since it's the only error returned.
Here we use the "advanced" mode of taking the error from the AST list, right?
(Just double checking understanding)

Reply:
It's checking both `ast.Errors[0]` and the  `err` returned by `Parse` against `ErrKeywordIsNotAllowed`

Reply:
Oh yeah, the reuse of `e` confused me. :+1: